File: src/main/solved/Programmers/Kakao_Blind_Recruitment_2022/신고 결과 받기.java
package Programmers.Kakao_Blind_Recruitment_2022;

import java.util.Arrays;
import java.util.HashSet;
import java.util.List;

class Solution {
    private HashSet<String> reportSet = new HashSet();
    public int[] solution(String[] id_list, String[] report, int k) {
        List<String> idList = Arrays.asList(id_list);
        int[] answer = new int[id_list.length];
        int[] userReportedCnt  = new int[id_list.length];

        for(String r : report) reportSet.add(r);
        for(String r : reportSet){
            String[] reportContents = r.split(" ");
            userReportedCnt[idList.indexOf(reportContents[1])] +=1;
        }
        for(String r : reportSet){
            String[] reportContents = r.split(" ");
            if(userReportedCnt[idList.indexOf(reportContents[1])] >= k){
                answer[idList.indexOf(reportContents[0])] +=1;
            }
        }
        return answer;
    }
}

File: src/main/solved/Programmers/level1/kotlin/신고 결과 받기.kt
class Solution {
    fun solution(id_list: Array<String>, report: Array<String>, k: Int): IntArray {
        var answer = IntArray(id_list.size)
        var reportCntList = IntArray(id_list.size)
        var idList = id_list.toList()
        var hashSet = HashSet<String>()

        for (r in report) hashSet.add(r)

        for (r in hashSet) {
            val str = r.split(" ")
            reportCntList[idList.indexOf(str[1])] +=1
        }

        for (r in hashSet) {
            val str = r.split(" ")
            if (reportCntList[idList.indexOf(str[1])] >= k) {
                answer[idList.indexOf(str[0])] += 1
            }
        }

        return answer
    }
}